### PR TITLE
Highlight primary category in menu

### PR DESCRIPTION
**Pull request by diedexx**
Sets submenuitems as 'active' if the post you're watching has the same _primary_ category as the label of the menuitem

### CODE REVIEW

File: php/class-yoast-navigation.php [hunk 1]
```diff
@@ -84,6 +84,21 @@ private function get_children_menu_data( $main_menu_item ) {
...
+	/**
+	 * Check if the submenu item is active.
+	 *
+	 * @param Menu_Item $sub_menu_item
+	 *
+	 * @return bool
+	 */
+	private function sub_menu_item_is_active( Menu_Item $sub_menu_item ) {
+		$primary_category    = yoast_get_primary_term( 'category', get_the_ID() );
+		$is_primary_category = strcasecmp( $sub_menu_item->getLabel() === $primary_category ) === 0;
```
Comment (moorscode) [inline php/class-yoast-navigation.php]: This is not going to work: `strcasecmp( $sub_menu_item->getLabel() === $primary_category )`
Have you tested this?

File: php/class-yoast-navigation.php [hunk 2]
```diff
@@ -84,6 +84,21 @@ private function get_children_menu_data( $main_menu_item ) {
...
 
+	/**
+	 * Check if the submenu item is active.
+	 *
+	 * @param Menu_Item $sub_menu_item
+	 *
+	 * @return bool
+	 */
+	private function sub_menu_item_is_active( Menu_Item $sub_menu_item ) {
+		$primary_category    = yoast_get_primary_term( 'category', get_the_ID() );
```
Comment (moorscode) [inline php/class-yoast-navigation.php]: `yoast_get_primary_term` wont be available everywhere

File: php/class-yoast-navigation.php [hunk 3]
```diff
@@ -84,6 +84,23 @@ private function get_children_menu_data( $main_menu_item ) {
...
+	 * @return bool
+	 */
+	private function menu_item_is_active( Menu_Item $sub_menu_item ) {
+		if ( function_exists( 'yoast_get_primary_term' ) ) {
+			$primary_category    = yoast_get_primary_term( 'category', get_the_ID() );
+			$is_primary_category = strcasecmp( $sub_menu_item->getLabel(), $primary_category ) === 0;
+		}
+		$is_same_url = ( $this->current_url === $sub_menu_item->getUrl() );
+
+		return $is_primary_category || $is_same_url;
```
Comment (moorscode) [inline php/class-yoast-navigation.php]: `$is_primary_category` is not always defined and will throw a notice.

File: php/class-yoast-navigation.php [hunk 4]
```diff
@@ -84,6 +84,23 @@ private function get_children_menu_data( $main_menu_item ) {
...
+	/**
+	 * Check if the child of a menu item is active.
+	 *
+	 * @param Menu_Item $sub_menu_item
+	 *
+	 * @return bool
+	 */
+	private function menu_item_is_active( Menu_Item $sub_menu_item ) {
+		if ( function_exists( 'yoast_get_primary_term' ) ) {
+			$primary_category    = yoast_get_primary_term( 'category', get_the_ID() );
```
Comment (moorscode) [inline php/class-yoast-navigation.php]: You don't have to pass `get_the_ID` but even safer would be to use `get_queried_object_id()`

File: html_includes/partials/navigation-menu-link.php [hunk 5]
```diff
@@ -11,15 +11,15 @@
 echo '<a href="' . $menu_item['url'] . '"' . $anchor_class . '>';
 
 if ( $type === 'main' ) {
-	echo $menu_item['label'];
+	echo esc_html( $menu_item['label'] );
 }
 
 if ( ! empty( $menu_item['icon'] ) ) {
 	echo '<span class="fa fa-' . $menu_item['icon'] . '" aria-hidden="true"></span>';
```
Comment (rensw90) [inline html_includes/partials/navigation-menu-link.php]: esc_html

**Comment by rensw90** [comment]
Cr-Done 👍

**Comment by rensw90** [comment]
Acceptance done 👍

**Pull request merged**